esp/GBPlay/main/hardware/wifi.c

```diff
@@ -187,12 +206,12 @@ bool wifi_connect(const char* ssid, const char* password, bool force)
             goto end;
         }
 
-        wifi_disconnect();
-
         ESP_ERROR_CHECK(esp_wifi_set_config(WIFI_IF_STA, &cfg));
 
         for (int i = 0; !did_connect && i < MAX_CONNECTION_RETRY_COUNT; ++i)
         {
+            _disconnect();
```

Comment:
Disconnect before every reconnect. We time out after 15 seconds of trying to connect, and don't want to reconnect while the Wi-Fi driver is still trying.